src/averageAges.js

```diff
@@ -56,6 +80,20 @@ function calculateWomenAverageAge(people, withChildren) {
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
   // write code here
+  const children = people.filter(child =>
+    onlyWithSon
+      ? people.find(mother => mother.name === child.mother)
```

Comment:
Instead find on every filter iteration create before filter object with method reduce
```
const peopleMap = people.reduce((acc, person) => {
    acc[person.name] = person;

    return acc;
  }, {});
  ````
  and instead find 
  ```
  peopleMap[p.mother]
  ````